app/view_models/base_view_model.rb

```diff
@@ -5,17 +5,17 @@ class BaseViewModel
   ID_FIELDS = ['id'].freeze
 
   class Builder
-    attr_reader :klass, :claim, :rows, :return_array
+    attr_reader :klass, :crime_application, :rows, :return_array
 
-    def initialize(class_type, claim, *nesting)
-      # TODO: Add mechanism to determine namespace dynamically
-      @klass = "NonStandardMagistratesPayment::V#{claim.json_schema_version}::#{class_type.to_s.camelcase}".constantize
-      @claim = claim
+    def initialize(class_type, crime_application, *nesting)
+      namespace = crime_application.is_a?(Claim) ? 'NonStandardMagistratesPayment' : 'PriorAuthority'
```

Comment:
would prefer to use the `application_type` in the check over the class - better yet move it into the `CrimeApplication` class so that it can return the appropriate namespace.

Reply:
Oh that's a great idea. Done.